packages/cli/src/controller/publish-controller.ts

```diff
@@ -61,24 +56,19 @@ export async function uploadToIpfs(
 
     validateCommonProjectManifest(schema);
 
-    const parsingFunctions = [
-      parseSubstrateProjectManifest,
-      parseCosmosProjectManifest,
-      parseAlgorandProjectManifest,
-      parseEthereumProjectManifest,
-      parseNearProjectManifest,
-      parseStellarProjectManifest,
-      parseConcordiumProjectManifest,
-    ];
-
-    let manifest = null;
-    for (const parseFunction of parsingFunctions) {
-      try {
-        manifest = parseFunction(schema).asImpl;
-        break; // Exit the loop if successful
-      } catch (e) {
-        // Continue to the next parsing function
-      }
+    const networkFamily = getProjectNetwork(schema);
+    const module = loadDependency(networkFamily);
+    const functionName = `parse${networkFamily}ProjectManifest`;
+    // @ts-ignore
+    // TODO,need fix type
+    const parseFunction = module[functionName];
```

Comment:
You might be able to fix this with https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html on the ModuleCache